### PR TITLE
BZ2107507: Added 4.11-specific note about checkup permissions

### PR DESCRIPTION
**Pull request by sjhala-ccs**
Version(s): 4.11

Issue: https://bugzilla.redhat.com/show_bug.cgi?id=2107507

Link to docs preview: http://file.rdu.redhat.com/sjhala/BZ2107507-4.11-update/virt/logging_events_monitoring/virt-running-cluster-checkups.html#virt-about-cluster-checkup-framework_virt-running-cluster-checkups

QE review:
- [x] QE has approved this change.

### CODE REVIEW

File: modules/virt-about-cluster-checkup-framework.adoc [hunk 1]
```diff
@@ -10,4 +10,9 @@ A _checkup_ is an automated test workload that allows you to verify if a specifi
...
 
-Running a predefined checkup in the cluster involves setting up the namespace and service account for the framework, creating the `ClusterRole` and `ClusterRoleBinding` objects for the service account, enabling permissions for the checkup, and creating the input config map and the checkup job. You can run a checkup multiple times. 
+Running a predefined checkup in the cluster involves setting up the namespace and service account for the framework, creating the `ClusterRole` and `ClusterRoleBinding` objects for the service account, enabling permissions for the checkup, and creating the input config map and the checkup job. You can run a checkup multiple times.
+
+[WARNING]
+====
+Before enabling permissions that the checkup requires for cluster access, you must verify that the checkup image is from a trustworthy source. This is because the framework automatically binds the `ClusterRole` objects provided by the checkup vendor to grant permissions to the checkup instance.
```
Comment (orelmisan) [inline modules/virt-about-cluster-checkup-framework.adoc:17]: What does "enabling permissions" mean?

**Review by orelmisan — COMMENTED** [review]
Thank you for the change @sjhala-ccs.
The message of the warning should be IMO:
1. Always verify the checkup's image before running it.
2. Always review the checkup's permissions before creating the ClusterRole object(s).
3. Always verify the names of ClusterRole objects you define in the user-supplied ConfigMap, because the framework automatically binds them to the checkup instance.

File: modules/virt-about-cluster-checkup-framework.adoc (hunk 1 above)
Comment (sjhala-ccs) [reply modules/virt-about-cluster-checkup-framework.adoc]: @orelmisan I have reworded the note per your suggestion.

**Review by orelmisan — APPROVED** [review]
Thank you for the changes @sjhala-ccs.

**Comment by ousleyp** [comment]
LGTM!

File: modules/virt-about-cluster-checkup-framework.adoc [hunk 2]
```diff
@@ -10,4 +10,13 @@ A _checkup_ is an automated test workload that allows you to verify if a specifi
...
 
-Running a predefined checkup in the cluster involves setting up the namespace and service account for the framework, creating the `ClusterRole` and `ClusterRoleBinding` objects for the service account, enabling permissions for the checkup, and creating the input config map and the checkup job. You can run a checkup multiple times. 
+Running a predefined checkup in the cluster involves setting up the namespace and service account for the framework, creating the `ClusterRole` and `ClusterRoleBinding` objects for the service account, enabling permissions for the checkup, and creating the input config map and the checkup job. You can run a checkup multiple times.
+
+[IMPORTANT]
+====
+You must always:
+
+* Verify that the checkup image is from a trustworthy source before applying it.
+* Review the checkup permissions before creating the `ClusterRole` objects.
```
Comment (yossisegev) [inline modules/virt-about-cluster-checkup-framework.adoc:20]: Either "ClusterRoleBinding" should be added ("before creating the `ClusterRole` and `ClusterRoleBinding`objects"), or change "objects" to "object" (AFAIR only one `ClusterRole` object is needed for the job).

**Review by yossisegev — CHANGES_REQUESTED** [review]

File: modules/virt-about-cluster-checkup-framework.adoc (hunk 2 above)
Comment (orelmisan) [reply modules/virt-about-cluster-checkup-framework.adoc:20]: In 4.11 the `checkup-framework` Job creates the `ClusterRoleBinding` objects automatically.
Users have to review the `ClusterRoleBinding` that they apply.
The phrasing is correct IMO.

Comment (yossisegev) [reply modules/virt-about-cluster-checkup-framework.adoc:20]: Very well, then LGTM.
Thanks for the clarification, Orel.

**Pull request merged**